### PR TITLE
fix: uri encode state part of fragment

### PR DESCRIPTION
**Pull request by danielgent**
I figure we merge this and then I'll keep running the login2 playwright tests :see_no_evil:

### CODE REVIEW

File: src/helpers/apply-token-response.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ function applyTokenResponseAsFragment(
   anchorLinks.set("token_type", "Bearer");
 
   if (state) {
-    anchorLinks.set("state", state);
+    anchorLinks.set("state", encodeURIComponent(state));
```
Comment (danielgent) [inline src/helpers/apply-token-response.ts:80]: we are not doing this on the query response helper. I wonder how this worked then? :thinking:

File: test/integration/flows/social.spec.ts [hunk 2]
```diff
@@ -124,7 +124,9 @@ describe("social sign on", () => {
         expect(socialCallbackQuery2.get("access_token")).toBeDefined();
         expect(socialCallbackQuery2.get("id_token")).toBeDefined();
         expect(socialCallbackQuery2.get("expires_in")).toBe("86400");
-        expect(socialCallbackQuery2.get("state")).toBe(LOGIN2_STATE);
+        expect(socialCallbackQuery2.get("state")).toBe(
+          encodeURIComponent(LOGIN2_STATE),
```
Comment (danielgent) [inline test/integration/flows/social.spec.ts:128]: this shows the change I've made

File: src/helpers/apply-token-response.ts (hunk 1 above)
Comment (markusahlstrand) [reply src/helpers/apply-token-response.ts:80]: You shouldn't need to do that right? That's the beauty of URLSearchParams I though?

Comment (danielgent) [reply src/helpers/apply-token-response.ts:80]: Ahhhh but below we're setting the to string of URLSearchParams to the hash of a redirect URL...

Does to string decode the params then? I could have a play around as I don't like making changes I don't understand

**Pull request merged**